pkg/rewards/claim.go

```diff
@@ -75,13 +80,142 @@ func getClaimFlags() []cli.Flag {
 		&ProofStoreBaseURLFlag,
 		&flags.VerboseFlag,
 		&flags.SilentFlag,
+		&flags.BatchClaimFile,
 	}
 
 	allFlags := append(baseFlags, flags.GetSignerFlags()...)
 	sort.Sort(cli.FlagsByName(allFlags))
 	return allFlags
 }
 
+func BatchClaim(
+	ctx context.Context,
+	logger logging.Logger,
+	ethClient *ethclient.Client,
+	elReader *elcontracts.ChainReader,
+	df *httpProofDataFetcher.HttpProofDataFetcher,
+	config *ClaimConfig,
+	p utils.Prompter,
+	claimDate string,
+	rootIndex uint32,
+	proofData *proofDataFetcher.RewardProofData,
+) error {
+
+	yamlFile, err := os.ReadFile(config.BatchClaimFile)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to read YAML config file", err)
+	}
+
+	var claimConfigs []struct {
+		EarnerAddress  string   `yaml:"earner_address"`
+		TokenAddresses []string `yaml:"token_addresses"`
+	}
+
+	err = yaml.Unmarshal(yamlFile, &claimConfigs)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to parse YAML config", err)
+	}
+
+	var elClaims []rewardscoordinator.IRewardsCoordinatorRewardsMerkleClaim
+	var claims []contractrewardscoordinator.IRewardsCoordinatorRewardsMerkleClaim
+	var accounts []merkletree.MerkleTree
+
+	for _, claimConfig := range claimConfigs {
+		earnerAddr := gethcommon.HexToAddress(claimConfig.EarnerAddress)
+
+		var tokenAddrs []gethcommon.Address
+
+		// Empty token addresses list will create a claim for all tokens claimable
+		// by the earner address.
+		if len(claimConfig.TokenAddresses) != 0 {
+			for _, addr := range claimConfig.TokenAddresses {
+				tokenAddrs = append(tokenAddrs, gethcommon.HexToAddress(addr))
+			}
+		}
+
+		elClaim, claim, account, err := generateClaimPayload(
+			ctx,
+			rootIndex,
+			proofData,
+			elReader,
+			logger,
+			earnerAddr,
+			tokenAddrs,
+		)
+
+		if err != nil {
+			logger.Warnf("Failed to process claim for earner %s: %v", earnerAddr.String(), err)
+			continue
+		}
+
+		elClaims = append(elClaims, *elClaim)
+		claims = append(claims, *claim)
+		accounts = append(accounts, *account)
+
+	}
+
+	err = broadcastClaims(config, ethClient, logger, p, ctx, &elClaims, &claims, &accounts)
+	return err
+}
+
+func generateClaimPayload(
+	ctx context.Context,
+	rootIndex uint32,
+	proofData *proofDataFetcher.RewardProofData,
+	elReader *elcontracts.ChainReader,
+	logger logging.Logger,
```

Comment:
major nit: small ick I get is when things which are contextually broader than the method being called are not first in method signatures. For example, logger is much broader context for this class than say the list of tokenAddresses. I would put it after `context` in this signature.

There's a topic on class cohesion which probably correlates here. Fields with higher cohesion (~broader context) go first in the method signatures before more specific parameters. 

By no-means a requirement, but I find this helps me quickly organize my thoughts on a method.

Reply:
will save this for a future refactor, there are also too many params in this method which would like to eventually fix